api/v1alpha1/lvmcluster_webhook.go

```diff
@@ -0,0 +1,266 @@
+			}
+		}
+
+		if deviceClass.DeviceSelector != nil {
+			newDevices = deviceClass.DeviceSelector.Paths
+		}
+
+		oldDevices, err = oldLVMCluster.getPathsOfDeviceClass(deviceClass.Name)
+
+		// if devices are removed now
+		if len(oldDevices) > len(newDevices) {
+			return fmt.Errorf("Invalid:devices can not be removed from the LVMCluster once added.")
+		}
+
+		// if devices are added now
+		if len(oldDevices) == 0 && len(newDevices) > 0 && err != ErrDeviceClassNotFound {
+			return fmt.Errorf("Invalid:devices can not be added in the LVMCluster once created without devices.")
+		}
+
+		deviceMap := make(map[string]bool)
+
+		for _, device := range oldDevices {
+			deviceMap[device] = true
+		}
+
+		for _, device := range newDevices {
+			delete(deviceMap, device)
+		}
+
+		// if any old device is removed now
+		if len(deviceMap) != 0 {
+			return fmt.Errorf("Invalid:some of devices are deleted from the LVMCluster. "+
+				"Device can not be removed from the LVMCluster once added. "+
+				"oldDevices:%s, newDevices:%s", oldDevices, newDevices)
+		}
+	}
+
+	return nil
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (l *LVMCluster) ValidateDelete() error {
+	lvmclusterlog.Info("validate delete", "name", l.Name)
+
+	return nil
+}
+
+func (l *LVMCluster) verifyPathsAreNotEmpty() error {
+
+	for _, deviceClass := range l.Spec.Storage.DeviceClasses {
+		if deviceClass.DeviceSelector != nil {
+			if len(deviceClass.DeviceSelector.Paths) == 0 {
+				return fmt.Errorf("Path list should not be empty when DeviceSelector is specified.")
+			}
+		}
+	}
+
+	return nil
+}
+
+func (l *LVMCluster) verifyAbsolutePath() error {
+
+	for _, deviceClass := range l.Spec.Storage.DeviceClasses {
+		if deviceClass.DeviceSelector != nil {
+			for _, path := range deviceClass.DeviceSelector.Paths {
+				if !strings.HasPrefix(path, "/dev/") {
+					return fmt.Errorf("Given path %s is not an absolute path. "+
+						"Please provide the absolute path to the device", path)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (l *LVMCluster) verifyNoDeviceOverlap() error {
+
+	// make sure no device overlap with another VGs
+	// use map to find the duplicate entries for paths
+	/*
+		{
+		  "nodeSelector1": {
+		        "/dev/sda": "vg1",
+		        "/dev/sdb": "vg1"
+		    },
+		    "nodeSelector2": {
+		        "/dev/sda": "vg1",
+		        "/dev/sdb": "vg1"
+		    }
+		}
+	*/
+	devices := make(map[string]map[string]string)
+
+	for _, deviceClass := range l.Spec.Storage.DeviceClasses {
+		if deviceClass.DeviceSelector != nil {
+			nodeSelector := deviceClass.NodeSelector.String()
+			for _, path := range deviceClass.DeviceSelector.Paths {
+				if val, ok := devices[nodeSelector][path]; ok {
+					var err error
+					if val != deviceClass.Name {
+						err = fmt.Errorf("Error: device path %s overlaps in two different deviceClasss %s and %s", path, val, deviceClass.Name)
```

Comment:
Ideally this is no required seeing that we are now restricting it to one deviceClass.